runtime/src/main/java/uk/ac/manchester/tornado/runtime/tasks/TornadoTaskSchedule.java

```diff
@@ -899,12 +899,11 @@ private void runScheduleWithParallelProfiler(Policy policy) {
 
         // One additional threads is reserved for sequential CPU execution
         final int numThreads = numDevices + 1;
-        final int indexSequential = numDevices;
         Thread[] threads = new Thread[numThreads];
         long[] totalTimers = new long[numThreads];
 
         // Last Thread runs the sequential code
-        runParallelSequential(policy, threads, indexSequential, timer, totalTimers);
+        runParallelSequential(policy, threads, numDevices, timer, totalTimers);
```

Comment:
This is not the number of devices. It is the index, within the array of all devices in which we keep a reference to the sequential execution. Please revert to `indexSequential`

Reply:
Sure, I will revert it but when I was going through the code, it seems we reassign `numDevices` to `indexSequential`.
`final int indexSequential = numDevices;`

Reply:
> Sure, I will revert it but when I was going through the code, it seems we reassign `numDevices` to `indexSequential`.
> `final int indexSequential = numDevices;`

Hi @Gaur4vGaur, the variable `indexSequential` is the index of the sequential thread that resides in the matrix of threads. If we have _n_ devices the number of threads is _n+1_, because it includes the sequential execution. In our case the last index (_n_) is used for the sequential thread, as the first _n_ indexes (_0-(n-1)_) are used for the threads that are associated with the _n_ devices.